bigtable/client/table.h

```diff
@@ -220,6 +224,57 @@ class Table {
                          std::vector<Mutation> true_mutations,
                          std::vector<Mutation> false_mutations);
 
+  using RpcUtils = bigtable::internal::UnaryRpcUtils<DataClient>;
+  using StubType = RpcUtils::StubType;
+
+  /**
+   * Read the row modify it according to ReadModifyRule and return the row.
+   *
+   * @tparam Args this is zero or more ReadModifyWriteRules to apply to the row
+   * @param row_key the row to read
+   * @param rule to modify the row. Two types of rules are applied here
+   *     AppendValue which will read the existing value and append the
+   *     text provided to the value.
+   *     IncrementAmount which will read the existing uint64 big-endian-int
+   *     and add the value provided.
+   *     Both rules accept the family and column identifier to modify.
+   * @returns modified row
+   */
+  template <typename... Args>
+  std::unique_ptr<Row> ReadModifyWriteRow(std::string row_key,
+                                          bigtable::ReadModifyWriteRule rule,
+                                          Args&&... rules) {
+    btproto::ReadModifyWriteRowRequest request;
+    request.set_table_name(table_name_);
+    request.set_row_key(std::move(row_key));
+
+    // This ugly thing provides a better compile-time error message than
+    // just letting the compiler figure things out 3 levels deep
+    // as it recurses on append_types().
```

Comment:
There is no recursion here, and certainly not on `append_types()`, so the comment should be different.  Maybe:

```C++
// Generate a better error message that the default one if the types do not match.
```